ui/selectors/assets.ts

```diff
@@ -40,3 +59,152 @@ export function getAssetsMetadata(state: AssetsState) {
 export function getAssetsRates(state: AssetsRatesState) {
   return state.metamask.conversionRates;
 }
+
+export const getTokenBalancesEvm = createDeepEqualSelector(
+  getSelectedAccountTokensAcrossChains,
+  getSelectedAccountNativeTokenCachedBalanceByChainId,
+  getTokenBalances,
+  (_state, accountAddress) => accountAddress,
+  getMarketData,
+  getCurrencyRates,
+  getPreferences,
+  getIsTokenNetworkFilterEqualCurrentNetwork,
+  getCurrentNetwork,
+  (
+    selectedAccountTokensChains,
+    nativeBalances,
+    tokenBalances,
+    selectedAccountAddress,
+    marketData,
+    currencyRates,
+    preferences,
+    isOnCurrentNetwork,
+    currentNetwork,
+  ) => {
+    const { hideZeroBalanceTokens } = preferences;
```

Comment:
Should we also add that logic to the non-evm selector ?

Reply:
That setting is introduced for evm tokens because the user is able to manually import tokens he has no balance for to his wallet. With that setting ON he is able to hide the tokens without balance.
The manual token import for NON-EVM is not yet supported; i think that is why the logic was not initially introduced in non-evm selector. I would expect that the assets we get from the snap once the account is added to all have a balance.
However, would be good to either add a TODO comment here to keep track of this of to add the preference even though it should have no effect in the selector for now